sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java

```diff
@@ -3,79 +3,89 @@
 /**
  * Auto-config to provide default {@link CredentialsProvider} for all Azure services
  *
  * @author Warren Zhu
  */
 @Configuration
 @EnableConfigurationProperties(AzureProperties.class)
-@ConditionalOnClass(Azure.class)
+@ConditionalOnClass(AzureResourceManager.class)
 @ConditionalOnProperty(prefix = "spring.cloud.azure", value = { "resource-group" })
 public class AzureContextAutoConfiguration {
 
-    private static final String PROJECT_VERSION =
-        AzureContextAutoConfiguration.class.getPackage().getImplementationVersion();
+    private static final String PROJECT_VERSION = AzureContextAutoConfiguration.class.getPackage()
+                                                                                     .getImplementationVersion();
     private static final String SPRING_CLOUD_USER_AGENT = "spring-cloud-azure/" + PROJECT_VERSION;
 
-    @Bean
-    @ConditionalOnMissingBean
-    public ResourceManagerProvider resourceManagerProvider(Azure azure, AzureProperties azureProperties) {
-        return new AzureResourceManagerProvider(azure, azureProperties);
-    }
 
     /**
-     * Create an {@link Azure} bean.
-     * @param credentials The credential to connect to Azure.
-     * @param azureProperties The configured Azure properties.
-     * @return An Azure object.
-     * @throws IOException When IOException happens.
+     * Create an {@link AzureResourceManager} bean.
+     *
+     * @param credential The credential to connect to Azure.
+     * @param profile The azure profile.
+     * @return An AzureResourceManager object.
      */
     @Bean
     @ConditionalOnMissingBean
-    public Azure azure(AzureTokenCredentials credentials, AzureProperties azureProperties) throws IOException {
-        RestClient restClient = new RestClient.Builder()
-            .withBaseUrl(credentials.environment(), AzureEnvironment.Endpoint.RESOURCE_MANAGER)
-            .withCredentials(credentials).withSerializerAdapter(new AzureJacksonAdapter())
-            .withResponseBuilderFactory(new AzureResponseBuilder.Factory())
-            .withInterceptor(new ProviderRegistrationInterceptor(credentials))
-            .withInterceptor(new ResourceManagerThrottlingInterceptor()).withUserAgent(SPRING_CLOUD_USER_AGENT)
-            .build();
+    public AzureResourceManager azureResourceManager(TokenCredential credential, AzureProfile profile) {
+        // TODO (xiada) USER AGENT
+        return AzureResourceManager.configure()
+//                                   .withLogLevel(HttpLogDetailLevel.BASIC)
+                                   .authenticate(credential, profile)
+                                   .withDefaultSubscription();
+    }
+//
+//    @VisibleForTesting
+//    protected Azure authenticateToAzure(RestClient restClient, String subscriptionId,
+//                                        AzureTokenCredentials credentials) {
+//        return Azure.authenticate(restClient, credentials.domain()).withSubscription(subscriptionId);
+//    }
 
```

Comment:
the code you have marked as comment here, I am not sure whether it's appropriate, should we delete?

Reply:
Yes, we should delete them.